src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -271,6 +271,7 @@ protected override void ContextUnloaded(ExecutionContext context)
                     {
                         ExecutionContextState contextState = CurrentContext.GetState<ExecutionContextState>();
                         contextState.NotificationCount += state.NotificationCount;
+                        if (contextState.RequireReturnValue) Push(StackItem.Null);
```

Comment:
I've implemented and tested this logic in neo-go and catched a bug, not sure if it's the same in C# implementation.

We don't set `RequireReturnValue` of parent's context to `false` after pushing `Null`. Consider the situation when two methods are sequentially called from parent's context: method A has no return value and method B has the one. After method A pushes its return value to stack, we still have `RequireReturnValue` set to true in the parent's context. Method B doesn't change `RequireReturnValue` of the parent's context, thus if no exception occurs, on method B's context unloading, we are pushing one extra `Null` on stack, which is wrong.

So I think we should explicitly set `contextState.RequireReturnValue` to `false` after this line.

Reply:
And, to be more presise, we should explicitly set `contextState.RequireReturnValue` to `false` even if `UncaughtException` is not null.